### PR TITLE
Move all fs calls to main process, invoke via IPC

### PR DESCRIPTION
**Pull request by joeyballentine**
This moves all the usage of FS out of the renderer.

One unfortunate thing I had to do was move a bit of code out of input-overrides, as that file imports from FS, so I moved the tiny bit that was used in the renderer out into its own file.

### CODE REVIEW

**Comment by joeyballentine** [comment]
I finally figured out why Vite wasn't working for me when I tried switching to it: it doesn't support nodejs stuff being called from the browser. And technically it's bad practice and a security concern to have that enabled in webpack, I just never cared.

So, since it's technically correct and I want to try out Vite, we need to do this. (It doesn't mean we actually will switch to Vite for sure, I just haven't gotten the opportunity to really try it out since it won't compile)

File: src/renderer/components/settings/components.tsx [hunk 1]
```diff
@@ -151,9 +155,11 @@ const CacheSetting = memo(({ setting, value, setValue }: SettingsProps<'cache'>)
                             locationPromise
                                 .then(async (cacheLocation) => {
                                     try {
-                                        await access(cacheLocation);
+                                        await ipcRenderer.invoke('fs-access', cacheLocation);
```
Comment (RunDevelopment) [inline src/renderer/components/settings/components.tsx:158]: This relies on `access` throwing an error. I don't think errors are ICP compatible, so how does this work now?

File: src/main/gui/main-window.ts [hunk 2]
```diff
@@ -195,6 +196,24 @@ const registerEventHandlerPreSetup = (
             })().catch(log.error);
         });
     }
+
+    // Handle filesystem
+    ipcMain.handle('fs-read-file', async (event, path) => fs.readFile(path, { encoding: 'utf8' }));
+    ipcMain.handle('fs-write-file', async (event, path, content) =>
+        fs.writeFile(path, content, { encoding: 'utf8' })
```
Comment (RunDevelopment) [inline src/main/gui/main-window.ts:203]: This doesn't make sense to me. If `content` can be a buffer, then how does that interact with `encoding: 'utf8'`? Will it ignore the encoding unless you pass in a string?

Comment (joeyballentine) [reply src/main/gui/main-window.ts:203]: Good point. I'll make the options passed through as well

File: src/renderer/components/settings/components.tsx (hunk 1 above)
Comment (joeyballentine) [reply src/renderer/components/settings/components.tsx:158]: Based on s a quick Google search it seems it handles errors fine now, but I'll try this out later to confirm

Comment (joeyballentine) [reply src/renderer/components/settings/components.tsx:158]: can confirm it does properly pass through errors.

**Review by RunDevelopment — APPROVED** [review]

**Pull request merged**